sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceUtils.scala

```diff
@@ -46,6 +46,11 @@ object DataSourceUtils extends PredicateHelper {
    * The key to use for storing partitionBy columns as options.
    */
   val PARTITIONING_COLUMNS_KEY = "__partition_columns"
+  /**
+   * The key to use for specifying partition overwrite mode when
+   * INSERT OVERWRITE a partitioned data source table.
+   */
+  val PARTITION_OVERWRITE_MODE = "partitionOverwriteMode"
```

Comment:
What is the difference between this `DataSourceUtils.PARTITION_OVERWRITE_MODE` and `SQLConf.PARTITION_OVERWRITE_MODE`?

Reply:
Basiclly, we will use `DataSourceUtils.PARTITION_OVERWRITE_MODE` if user set `partitionOverwriteMode` in table properties or options. If not, then we will read `SQLConf.PARTITION_OVERWRITE_MODE`.